distributed/src/main/java/com/orientechnologies/orient/core/db/OrientDBDistributed.java

```diff
@@ -178,6 +180,30 @@ public void drop(String name, String user, String password) {
     }
   }
 
+  private void checkDbAvailable(String name) {
+    if (OSystemDatabase.SYSTEM_DB_NAME.equals(name)) return;
+    ODistributedServerManager.DB_STATUS dbStatus =
+        plugin.getDatabaseStatus(plugin.getLocalNodeName(), name);
+    if (dbStatus != ODistributedServerManager.DB_STATUS.ONLINE
+        && dbStatus != ODistributedServerManager.DB_STATUS.BACKUP) {
+      throw new OOfflineNodeException(
```

Comment:
Not sure whether we should use this exception or ODistributedRedirectException.
Looking at the client, I'd say both have problems:
- ODistributedRedirectException needs an explicit node to redirect to, that could be challenging
- OOfflineNodeException removes this node from the server list, not sure it's what we want, since it's a temporary situation that will recover by itself very soon

Reply:
Yes, I think the problem with redirent is that you need to provide an alternative, which in this case doesn't apply. For the offline exception, if you check the ConnectionStrategies test, I think it is covered. Meaning, once the DB is online, a new config is published and the clients gen update.

Reply:
Even if the node is removed from the list, should be re-added when a notification from the distributed make the node available, or when there are no-more node left to connect to. So should be fine to return the  OOfflineNodeException